src/test/java/io/neonbee/test/helper/MultipartResponseTest.java

```diff
@@ -0,0 +1,117 @@
+package io.neonbee.test.helper;
+
+import static com.google.common.collect.Iterables.get;
+import static com.google.common.truth.Truth.assertThat;
+import static io.vertx.core.MultiMap.caseInsensitiveMultiMap;
+import static io.vertx.core.buffer.Buffer.buffer;
+import static java.util.Collections.emptyList;
+import static java.util.Optional.ofNullable;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.ext.web.client.HttpResponse;
+
+class MultipartResponseTest {
```

Comment:
Before I look into this tests, I'd like to clarify if we can reuse ODataResponse.